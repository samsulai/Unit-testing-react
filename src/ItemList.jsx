import React from 'react'

const ItemList = ({  items }) => (

<ul>
{items.map((item) => (
<li key={item.id}>{item.name}</li>
	))}
</ul>
		
	
)
export default ItemList