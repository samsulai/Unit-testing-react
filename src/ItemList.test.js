import TestRender from "react-test-renderer"
import ItemList from "./ItemList"

describe("ItemList", () => {
	const items = [
{id : 1, name : "Item 1"},
{id : 2, name : "Item 2"},
{id : 3, name : "Item 3"}
	]
	it('renders correctly', () => {
 const tree = TestRender.create(<ItemList items={items} />)
 	expect(tree.toJSON()).toMatchSnapshot()
	})

it('renders correct number of items', () => {
 const tree = TestRender.create(<ItemList items={items} />)
 	const root = tree.root 
 	expect(root.findAllByType("li").length).toEqual(items.length)
	})
	
	it('renders the correct item names', () => {
 const tree = TestRender.create(<ItemList items={items} />)
 	const root = tree.root 
 	const liNodes = root.findAllByType('li')
 	liNodes.forEach((node, index)=> {
expect(node.children).toEqual([items[index].name])
 	})
 	
	})
})