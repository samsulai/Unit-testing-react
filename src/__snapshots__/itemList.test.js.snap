// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList renders correctly 1`] = `
<ul>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
  <li>
    Item 3
  </li>
</ul>
`;

exports[`itemList renders correctly 1`] = `
<itemList
  items={
    [
      {
        "id": 1,
        "name": "Item 1",
      },
      {
        "id": 2,
        "name": "Item 2",
      },
      {
        "id": 3,
        "name": "Item 3",
      },
    ]
  }
/>
`;
